### DISCUSSION TITLE
Workaround for passing task command line options to pre tasks?

### DISCUSSION THREAD

**Discussion by rberger** [pyinvoke/invoke#886] [Q&A]
From looking at issues history, I see that it's been a long time coming to have a way to pass task command line options to the `pre` tasks. There have been passing mentions of ways to possibly do this with context as a work around.

Wondering if there is any suggested (and hopefully documented) workarounds to accomplish passing this to pre tasks? It's really become a complete bottleneck for my use case.

The only solution I see so far is to switch to doing everything with `post` instead of `pre` and have the task getting the command line option to push it into  a global. This path is a bummer in multiple dimensions.

Also, does anyone read these Discussions? Or am I wasting my time posting here?

**Comment by rberger** [comment]
Another thing I wish I could do is have a command line option to tell a run to just execute the task specified and not call the pre or post tasks

**Comment by meeuw** [comment]
As a workaround I think you could use environment variables or a configuration file instead of command line options.

If you insist on command line options one could use the context to set and reuse the command line options in pre tasks. This should be done from another task which calls the subject task as a post task.

**Reply by rberger** [reply]
Thanks for the response.

I ended up refactoring to not use `pre` and just called the individual tasks in a top level meta task.

So now the individual sub tasks can be called with command line options and the meta task can pass the command line options to the sub tasks
